File: 백준/Bronze/5622. 다이얼/다이얼.cc
#include <iostream>
#include <string>

using namespace std;

int getTime(string sArr[], int size, char find) {
    for (int i = 0; i < size; i++) {
        if (sArr[i].find(find) != string::npos) {
            return i += 2;
        }
    }
    return 11;
}

int main() {
    string sArr[8] = {"ABC", "DEF", "GHI", "JKL", "MNO", "PQRS", "TUV", "WXYZ"};
    int size = sizeof(sArr) / sizeof(string);

    string line;
    cin >> line;

    int sum = 0;
    for (int i = 0; i < line.length(); i++) {
        char c = line.at(i);
        sum += getTime(sArr, size, c);
    }

    cout << sum + line.size();
}

File: 백준/Bronze/5622. 다이얼/다이얼.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    static int[] arr = new int[26];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        init();

        int sum = 0;
        for (char c : br.readLine().toCharArray()) {
            sum += arr[c - 'A'];
        }
        System.out.println(sum);
    }

    static void init() {
        String[] alphabets = {"ABC", "DEF", "GHI", "JKL", "MNO", "PQRS", "TUV", "WXYZ"};
        int number = 3;
        for (String alphabet : alphabets) {
            for (char c : alphabet.toCharArray()) {
                int idx = c - 'A';
                arr[idx] = number;
            }
            number++;
        }
    }
}